pkg/terraform/build_test.go

```diff
@@ -6,35 +6,44 @@ import (
 	"io/ioutil"
 	"testing"
 
+	"get.porter.sh/mixin/terraform/pkg"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
 func TestMixin_Build(t *testing.T) {
-	t.Run("build with the default Terraform version", func(t *testing.T) {
-		m := NewTestMixin(t)
-
-		ctx := context.Background()
-		err := m.Build(ctx)
-		require.NoError(t, err)
-
-		gotOutput := m.TestContext.GetOutput()
-		assert.Contains(t, gotOutput, "https://releases.hashicorp.com/terraform/1.2.9/terraform_1.2.9_linux_amd64.zip")
-		assert.NotContains(t, "{{.", gotOutput, "Not all of the template values were consumed")
-	})
-
-	t.Run("build with custom Terrafrom version", func(t *testing.T) {
-		b, err := ioutil.ReadFile("testdata/build-input-with-version.yaml")
-		require.NoError(t, err)
-
-		ctx := context.Background()
-		m := NewTestMixin(t)
-		m.In = bytes.NewReader(b)
-		err = m.Build(ctx)
-		require.NoError(t, err)
-
-		gotOutput := m.TestContext.GetOutput()
-		assert.Contains(t, gotOutput, "https://releases.hashicorp.com/terraform/0.13.0-rc1/terraform_0.13.0-rc1_linux_amd64.zip")
-		assert.NotContains(t, "{{.", gotOutput, "Not all of the template values were consumed")
-	})
+	testcases := []struct {
+		name              string
+		inputFile         string
+		expectedVersion   string
+		expectedUserAgent string
+	}{
+		{name: "build with custom config", inputFile: "testdata/build-input-with-config.yaml", expectedVersion: "https://releases.hashicorp.com/terraform/0.13.0-rc1/terraform_0.13.0-rc1_linux_amd64.zip", expectedUserAgent: "ENV PORTER_TERRAFORM_MIXIN_USER_AGENT_OPT_OUT=\"true\"\nENV AZURE_HTTP_USER_AGENT=\"\""},
+		{name: "build with the default Terraform config", expectedVersion: "https://releases.hashicorp.com/terraform/1.2.9/terraform_1.2.9_linux_amd64.zip", expectedUserAgent: "ENV PORTER_TERRAFORM_MIXIN_USER_AGENT_OPT_OUT=\"false\"\nENV AZURE_HTTP_USER_AGENT=\""},
```

Comment:
Shouldn't the `AZURE_HTTP_USER_AGENT` environment variable be set in the dockerfile for this test case?

Reply:
Oops, just fixed it